<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

## Footnotes

# Exercise 1

In [5]:
while True:
    print('*', end=' ')

* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * 

KeyboardInterrupt: 

# Exercise 2

In [1]:
#print = 'PRINT'
print('Hello World')

Hello World


# Exercise 3

In [4]:
%%timeit
squares = [i**2 for i in range(10)]
#print(f"squares: {squares}")

160 ns ± 1.16 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)


"7 runs, 10,000,000 loops each" means that the code was run for 7 rounds with the code being run 10 million times for each round. Each round took an average of 160 nanoseconds to run with the standard deviation of all the timings being 1.16 nanoseconds.

In [5]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)
#print(f"squares: {squares}")    

205 ns ± 1.7 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


"7 runs, 1,000,000 loops each" means that the code was run for 7 rounds with the code being run 1 million times for each round. Each round took an average of 205 nanoseconds to run with the standard deviation of all the timings being 1.7 nanoseconds.